src/openrct2/core/Json.cpp

```diff
@@ -26,42 +25,101 @@ namespace Json
             throw IOException("Json file too large.");
         }
 
-        utf8* fileData = Memory::Allocate<utf8>(fileLength + 1);
-        fs.Read(fileData, fileLength);
-        fileData[fileLength] = '\0';
+        auto fileData = std::string(static_cast<size_t>(fileLength) + 1, '\0');
+        fs.Read(static_cast<void*>(fileData.data()), fileLength);
 
-        json_error_t jsonLoadError;
-        json = json_loads(fileData, 0, &jsonLoadError);
-        Memory::Free(fileData);
+        json_t json;
 
-        if (json == nullptr)
+        try
         {
-            throw JsonException(&jsonLoadError);
+            json = json_t::parse(fileData, nullptr, true, true);
```

Comment:
Any reason why you have specified all optional elements with their defaults?

Asking because specifying the fourth argument causes it to fail compilation on Ubuntu 20.04, using nlohmann/json version 3.7.3-1.